Webapp/src/pages/DashboardPage.tsx

```diff
@@ -17,8 +17,52 @@ import HotelIcon from '@mui/icons-material/Hotel';
 import RepeatIcon from '@mui/icons-material/Repeat';
 import { TextField } from '@mui/material';
 import Switch from '@mui/material/Switch';
+import dayjs, { Dayjs } from 'dayjs';
+import Badge from '@mui/material/Badge';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { PickersDay, PickersDayProps } from '@mui/x-date-pickers/PickersDay';
+import { DateCalendar } from '@mui/x-date-pickers/DateCalendar';
+import { DayCalendarSkeleton } from '@mui/x-date-pickers/DayCalendarSkeleton';
 
+function getRandomNumber(min: number, max: number) {
+    return Math.round(Math.random() * (max - min) + min);
+}
+
+function fakeFetch(date: Dayjs, { signal }: { signal: AbortSignal }) {
+    return new Promise<{ daysToHighlight: number[] }>((resolve, reject) => {
+        const timeout = setTimeout(() => {
+            const daysInMonth = date.daysInMonth();
+            const daysToHighlight = [1, 2, 3].map(() => getRandomNumber(1, daysInMonth));
+
+            resolve({ daysToHighlight });
+        }, 500);
+
+        signal.onabort = () => {
+            clearTimeout(timeout);
+            reject(new DOMException('aborted', 'AbortError'));
+        };
+    });
+}
 
+const initialValue = dayjs('2024-11-18');
+
+function ServerDay(props: PickersDayProps<Dayjs> & { highlightedDays?: number[] }) {
+    const { highlightedDays = [], day, outsideCurrentMonth, ...other } = props;
```

Comment:
decompose these fields from the object, if there is no `highlightedDays` prop in the list, use a default `[]`